src/app/models/user.rb

```diff
@@ -8,6 +8,22 @@ class User < ApplicationRecord
   has_many :likes, dependent: :destroy
   has_many :saves, dependent: :destroy
 
+  has_many :follower_relationships, foreign_key: :following_id, class_name: 'Follow',
+                                    dependent: :destroy, inverse_of: :follower
+  has_many :followers, through: :follower_relationships, source: :follower
+
+  has_many :following_relationships, foreign_key: :follower_id, class_name: 'Follow',
+                                     dependent: :destroy, inverse_of: :following
+  has_many :following, through: :following_relationships, source: :following
```

Comment:
Instead of using has_many 4 times I think you can replace it with `has_and_belongs_to_many` twice, and you won't need the new model `Follow` because the join table doesn't have a third column, and in that case you won't need the `follows` table to have an ID because the relation identifies the record.
https://devdocs.io/rails~6.0/activerecord/associations/classmethods#method-i-has_and_belongs_to_many

Reply:
rubocop prefers `has_many :through` over `has_and_belongs_to_many` (and me also), I think it is stronger in definition than `has_and_belongs_to_many` association, because you can see the actual components of the association. What do you think?

Reply:
I see, I tend to follow rubocop, so I'll have to agree with your approach.